tool/release/CreateNotes.kt

```diff
@@ -0,0 +1,216 @@
+                .buildGetRequest(GenericUrl("https://api.github.com/repos/$org/$repo/commits/$commit"))
+                .setHeaders(
+                    HttpHeaders().setAuthorization("Token $githubToken").setAccept("application/vnd.github.v3+json")
+                )
+                .execute()
+            val body = Json.parse(String(response.content.readBytes()))
+            val notes = listOf(
+                CommitNote(
+                    title = body.asObject().get("commit").asObject().get("message").asString(),
+                    desc = "",
+                    type = CommitNoteType.OTHER
+                )
+            )
+            notes
+        }
+    }
+}
+
+fun createCommitNote(note: CommitNote): String {
+    var desc = StringBuilder()
+    var header = 0
+    for (line in note.desc.lines()) {
+        if (line.startsWith("##")) {
+            header += 1
+        } else if (header == 1) {
+            desc.append(line)
+        } else if (header > 1) {
+            break
+        }
+    }
+    return "- **${note.title}**\n  $desc"
+}
+
+fun createReleaseNote(commitNotes: List<CommitNote>, releaseTemplateFile: Path) {
+    val features = mutableListOf<CommitNote>()
+    val bugs = mutableListOf<CommitNote>()
+    val refactors = mutableListOf<CommitNote>()
+    val others = mutableListOf<CommitNote>()
+
+    commitNotes.forEach { note ->
+        when (note.type) {
+            CommitNoteType.FEATURE -> features.add(note)
+            CommitNoteType.BUG -> bugs.add(note)
+            CommitNoteType.REFACTOR -> refactors.add(note)
+            else -> others.add(note)
+        }
+    }
+
+    val releaseNote = """
+Install & Run: http://docs.vaticle.com/docs/running-typedb/install-and-run
+
+## New Features
+${features.map { feature -> createCommitNote(feature) }.joinToString("\n")}
+
+## Bugs Fixed
+${bugs.map { bug -> createCommitNote(bug) }.joinToString("\n")}
+
+## Code Refactor
+${refactors.map { refactor -> createCommitNote(refactor) }.joinToString("\n")}
+
+## Other Improvements
+${others.map { other -> createCommitNote(other) }.joinToString("\n")}
+    """
+    Files.write(releaseTemplateFile, releaseNote.toByteArray(UTF_8))
+}
+
+fun main(args: Array<String>) {
+    val buildWorkspaceDirEnv = "BUILD_WORKSPACE_DIRECTORY"
+    if (System.getenv(buildWorkspaceDirEnv) == null) throw RuntimeException("Not running from within Bazel workspace")
+    val workspaceDirectory = System.getenv(buildWorkspaceDirEnv)
+
+    val createTokenEnv = "CREATE_NOTES_TOKEN"
+    if (System.getenv(createTokenEnv) == null) throw RuntimeException("$createTokenEnv environment variable is not set")
+    val githubToken = System.getenv(createTokenEnv)
+
+    if (args.size != 5) throw RuntimeException("repository, version, commit, and release-template must be supplied")
+    val org = args[0]
+    val repo = args[1]
+    val version = args[2]
+    val toBeReleased = args[3]
+    val releaseTemplateFile = Paths.get(workspaceDirectory, args[4])
```

Comment:
@lolski destructuring syntax could probably be used here

```suggestion
    val (org, repo, version, toBeReleased, releaseTemplatePath) = args
```

Reply:
Fixed in fa97b42f13937eea3b1184fad7796e2349b39d46.